### PR TITLE
Cardano Node 1.35.6 aka Single Relay P2P release

### PR DESCRIPTION
**Pull request by coot**
Update `ouroboros-network-0.2.0.0`.

This also cherry-picked CI commits:
* e745ca869 from #4184
* #4513
* #4526
* #4590

### CODE REVIEW

**Comment by daehan-koreapool** [comment]
@coot 
Hi Marcin, 

In the `spo-announcements` discord channel, you've informed P2P relay should be included in the registration cert.
As a SPO, I've been running multiple hidden relays for security reasons.

Does this mean all the hidden relays need to be exposed publicly by including their DNS records in pool certs?

If not, would you be able to elaborate a bit more on what you mean by

>We would like to ask you to double check that your registration certificates contains valid domain names, otherwise other relays will not be able to communicate with your relays, and thus your blocks will not be delivered through the P2P relay.

**Comment by coot** [comment]
@daehan-koreapool it's ok to run hidden relays; as with non-P2P mode their downstream peers need to be explicitly configured to connect to your hidden relay(s).

The reminder is just to double check that your on chain registered relays have valid DNS names.

File: cardano-node/cardano-node.cabal [hunk 1]
```diff
@@ -1,7 +1,7 @@
 cabal-version: 3.0
 
 name:                   cardano-node
-version:                1.35.4
+version:                1.35.5
```
Comment (Jimbo4350) [inline cardano-node/cardano-node.cabal:4]: Shouldn't we be bumping all the packages cc: @disassembler

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
Minor changes. There are some commits that can be squashed together to clean up the git history.

File: cardano-node/cardano-node.cabal (hunk 1 above)
Comment (coot) [reply cardano-node/cardano-node.cabal:4]: Done

**Comment by coot** [comment]
I had to pin all `ledger` packages, since new versions where published to `CHaP` and I need a newer `CHaP` to get newer versions of `ouroboros-network` & `ouroboros-consensus`.

**Comment by andreabedini** [comment]
> I had to pin all `ledger` packages, since new versions where published to `CHaP` and I need a newer `CHaP` to get newer versions of `ouroboros-network` & `ouroboros-consensus`.

These are direct depedencies and you should put the constraint in the cabal file itself, not in the cabal.project. Rememeber that cabal.project is not published only (e.g.) cardano-api.cabal is.

**Comment by coot** [comment]
The haddock build is fixed in #4662.

**Comment by coot** [comment]
> These are direct depedencies and you should put the constraint in the cabal file itself, not in the cabal.project. Rememeber that cabal.project is not published only (e.g.) cardano-api.cabal is.

I agree :100:, although I think a suitable solution should be found by the `node` team in `master`.  @Jimbo4350 @JaredCorduan what do you think?

`network` packages are pinned in `*.cabal` files, although it can be improved.

**Comment by coot** [comment]
@andreabedini I opened an issue for this #4663.

**Comment by gitmachtl** [comment]
Will this release support the newer (reduced) p2p-topology syntax?

**Comment by coot** [comment]
@gitmachtl yes, `1.35.5` supports new (as well as old) topology format.

**Comment by coot** [comment]
I rebased the PR on recent changes merged to the `release/1.35` branch.

**Comment by coot** [comment]
The cicero job exited with:
```
linux error: flake 'github:input-output-hk/cardano-node/be70c4f951b4adff0789905d19cca466bf8c6c76' does not provide attribute 'packages.x86_64-linux.hydraJobsPr.linux.required', 'legacyPackages.x86_64-linux.hydraJobsPr.linux.required' or 'hydraJobsPr.linux.required'
```

**Comment by coot** [comment]
Requires #4769

**Comment by jbgi** [comment]
bors try

**Comment by disassembler** [comment]
bors try

**Comment by coot** [comment]
bors merge

**Pull request merged**